cmd/dagger/flags.go

```diff
@@ -317,8 +317,13 @@ func (v *directoryValue) Get(ctx context.Context, dag *dagger.Client, modSrc *da
 	// Try parsing as a Git URL
 	gitURL, err := parseGitURL(v.String())
 	if err == nil {
-		// TODO: use modArg.Ignore if not empty
-		return makeGitDirectory(gitURL, dag), nil
+		return dag.Directory().
+			WithDirectory(
+				"/",
+				makeGitDirectory(gitURL, dag),
+				dagger.DirectoryWithDirectoryOpts{
+					Exclude: modArg.Ignore,
+				}), nil
```

Comment:
This always makes a copy even if there's no ignore filters. Should only do it if Ignore isn't empty.